lib/Frontend/Frontend.cpp

```diff
@@ -478,8 +478,23 @@ void CompilerInstance::setupOutputBackend() {
         *CAS, *ResultCache, *CompileJobBaseKey,
         Invocation.getFrontendOptions().InputsAndOutputs,
         Invocation.getFrontendOptions().RequestedAction);
+
+    auto FilterBackend = llvm::vfs::makeFilteringOutputBackend(
+        CASOutputBackend,
+        [&](StringRef Path, std::optional<llvm::vfs::OutputConfig> Config) {
+          auto &InAndOuts = Invocation.getFrontendOptions().InputsAndOutputs;
+          if (InAndOuts.getPrincipalOutputType() != file_types::ID::TY_Object)
+            return true;
+          std::vector<std::string> OutputFiles =
```

Comment:
I will compute this beforehand, capture by value in the lambda so it doesn't get recomputed every time an output needs to be computed. Maybe even make this a set so you don't need to iterate through.